### DISCUSSION TITLE
Which set of pars should I choose? The one with lowest error or the one near the end of iterations?

### DISCUSSION THREAD

**Discussion by 3zhang** [optuna/optuna#4730] [Q&A]
Currently I'm using optuna to tune my lightgbm model. I've done 5000 trials with CmaEsSampler and this is the objective(rmse) history. The best trial is one ~2500. However there are other competitive trials near the end of iterations with only slightly higher errors. Which one should I use?
<img width="846" alt="newplot (9)" src="https://github.com/optuna/optuna/assets/8787897/9eb63d32-9257-43c6-b927-f18dca241c8f">

**Comment by nzw0301** [comment]
It depends on your problem. Usually, we use the model that achieved the lowest objective value because we can choose it automatically and its info is stored as `best_trial` in `study`. Of course, we can try other trials to improve the performance, for example, with a majority vote.

**Reply by 3zhang** [reply]
Could you elaborate on "majority vote"?

**Reply by nzw0301** [reply]
https://rasbt.github.io/mlxtend/user_guide/classifier/EnsembleVoteClassifier/
